runtime/common/MeasureCounts.cpp

```diff
@@ -88,31 +112,20 @@ std::vector<std::size_t> ExecutionResult::serialize() const {
     retData.push_back(bits.length());
     retData.push_back(count);
   }
-
   return retData;
 }
 
 void ExecutionResult::deserialize(std::vector<std::size_t> &data) {
   std::size_t stride = 0;
   while (stride < data.size()) {
-    auto nChars = data[stride];
-    stride++;
-    std::string name = "";
-    for (std::size_t i = 0; i < nChars; i++)
-      name += std::string(1, char(data[stride + i]));
+    std::string name = extractNameFromData(data, stride);
 
-    stride += nChars;
     std::unordered_map<std::string, std::size_t> localCounts;
-    auto nBs = data[stride];
-    stride++;
-    for (std::size_t j = stride; j < stride + nBs * 3; j += 3) {
-      auto bitstring_as_long = data[j];
-      auto size_of_bitstring = data[j + 1];
-      auto count = data[j + 2];
-      auto bs = longToBitString(size_of_bitstring, bitstring_as_long);
-      counts.insert({bs, count});
+    deserializeCounts(data, stride, localCounts);
+
+    for (const auto &entry : localCounts) {
+      counts.insert_or_assign(entry.first, entry.second);
```

Comment:
Why the call to `insert_or_assign` here vs just `insert`? It seems like a potential change in behavior. Is that intentional?